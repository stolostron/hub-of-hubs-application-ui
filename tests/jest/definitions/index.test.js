/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright Contributors to the Open Cluster Management project
import getResourceData, {
  getPrimaryKey,
  getSecondaryKey,
  getURIKey,
  getDefaultSearchField,
  getDefaultSortField,
  getTableKeys,
  getLink
} from "../../../src-web/definitions/index";

// regular properties/selector/reducer testing
// compare the received and expect values
describe("definitions/index tests", () => {
  it("should return the resource data for specified resource", () => {
    const item = {
      list: "HCMApplicationList",
      name: "HCMApplication"
    };
    expect(getResourceData(item)).toMatchSnapshot();
  });

  it("should return the primary key of node", () => {
    const item = {
      list: "HCMApplicationList",
      name: "HCMApplication"
    };
    expect(getPrimaryKey(item)).toMatchSnapshot();
  });

  it("should return the secondary key of node", () => {
    const item = {
      list: "HCMApplicationList",
      name: "HCMApplication"
    };
    expect(getSecondaryKey(item)).toMatchSnapshot();
  });

  it("should return the uri key of node", () => {
    const item = {
      list: "HCMApplicationList",
      name: "HCMApplication"
    };
    expect(getURIKey(item)).toMatchSnapshot();
  });

  it("should return the table keys for this node", () => {
    const item = {
      list: "HCMApplicationList",
      name: "HCMApplication"
    };
    expect(getTableKeys(item)).toMatchSnapshot();
  });

  it("should return the default search field of node", () => {
    const item = {
      list: "HCMApplicationList",
      name: "HCMApplication"
    };
    expect(getDefaultSearchField(item)).toMatchSnapshot();
  });

  it("should return the default sort field of node", () => {
    const item = {
      list: "HCMApplicationList",
      name: "HCMApplication"
    };
    expect(getDefaultSortField(item)).toMatchSnapshot();
  });

  it("should return the link of node as /namespace/name", () => {
    const link = true;
    const resource = {
      metadata: {
        namespace: "namespace",
        name: "name"
      }
    };
    expect(getLink(link, resource)).toMatchSnapshot();
  });

  it("should return the link of node as /domain/host", () => {
    const link = "domain/host";
    const resource = {
      domain: "domain",
      host: "host",
      metadata: {
        namespace: "namespace",
        name: "name"
      }
    };
    expect(getLink(link, resource)).toMatchSnapshot();
  });
});
