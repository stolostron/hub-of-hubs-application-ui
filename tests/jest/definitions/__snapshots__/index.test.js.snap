// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`definitions/index tests should return the default search field of node 1`] = `"name"`;

exports[`definitions/index tests should return the default sort field of node 1`] = `"name"`;

exports[`definitions/index tests should return the link of node as /domain/host 1`] = `"/domain/host"`;

exports[`definitions/index tests should return the link of node as /namespace/name 1`] = `"/namespace/name"`;

exports[`definitions/index tests should return the primary key of node 1`] = `"_uid"`;

exports[`definitions/index tests should return the resource data for specified resource 1`] = `
Object {
  "defaultSortField": "name",
  "emptyMessage": [Function],
  "emptyTitle": [Function],
  "groupFn": [Function],
  "groupSummaryFn": [Function],
  "keyFn": [Function],
  "pluralKey": "table.plural.application",
  "primaryKey": "_uid",
  "tableActionsResolver": [Function],
  "tableDropdown": Object {
    "actions": Array [
      Object {
        "msgKey": "application.type.acm",
        "path": "undefined/create",
      },
      Object {
        "label": "creation.app.section.techPreview",
        "labelColor": "creation.app.section.techPreview.color",
        "msgKey": "application.type.argo",
        "path": "undefined/argoappset",
      },
    ],
    "disableMsgKey": "actions.create.application.access.denied",
    "msgKey": "actions.create.application",
  },
  "tableFilter": Object {
    "labelKey": "table.filter.type.acm.application.label",
    "options": Array [
      Object {
        "labelKey": "table.filter.type.acm.application",
        "valueKey": "table.filter.type.acm.application.value",
      },
      Object {
        "labelKey": "table.filter.type.argo.application",
        "valueKey": "table.filter.type.argo.application.value",
      },
    ],
    "tableFilterFn": [Function],
  },
  "tableKeys": Array [
    Object {
      "msgKey": "table.header.name",
      "resourceKey": "name",
      "textFunction": [Function],
      "transformFunction": [Function],
      "transforms": Array [
        [Function],
      ],
    },
    Object {
      "disabled": [Function],
      "msgKey": "table.header.type",
      "resourceKey": "type",
      "textFunction": [Function],
      "transformFunction": [Function],
      "transforms": Array [
        [Function],
      ],
    },
    Object {
      "msgKey": "table.header.namespace",
      "resourceKey": "namespace",
      "textFunction": [Function],
      "tooltipKey": "table.header.application.namespace.tooltip",
      "transformFunction": [Function],
      "transforms": Array [
        [Function],
      ],
    },
    Object {
      "disabled": [Function],
      "msgKey": "table.header.clusters",
      "resourceKey": "clusterCount",
      "textFunction": [Function],
      "tooltipKey": "table.header.application.clusters.tooltip",
      "transformFunction": [Function],
    },
    Object {
      "disabled": [Function],
      "msgKey": "table.header.resource",
      "resourceKey": "hubChannels",
      "textFunction": [Function],
      "tooltipKey": "table.header.application.resource.tooltip",
      "transformFunction": [Function],
    },
    Object {
      "disabled": [Function],
      "msgKey": "table.header.timeWindow",
      "resourceKey": "hubSubscriptions",
      "textFunction": [Function],
      "tooltipKey": "table.header.application.timeWindow.tooltip",
      "transformFunction": [Function],
    },
    Object {
      "msgKey": "table.header.created",
      "resourceKey": "created",
      "transformFunction": [Function],
    },
  ],
  "uriKey": "name",
}
`;

exports[`definitions/index tests should return the secondary key of node 1`] = `"cluster"`;

exports[`definitions/index tests should return the table keys for this node 1`] = `
Array [
  Object {
    "msgKey": "table.header.name",
    "resourceKey": "name",
    "textFunction": [Function],
    "transformFunction": [Function],
    "transforms": Array [
      [Function],
    ],
  },
  Object {
    "disabled": [Function],
    "msgKey": "table.header.type",
    "resourceKey": "type",
    "textFunction": [Function],
    "transformFunction": [Function],
    "transforms": Array [
      [Function],
    ],
  },
  Object {
    "msgKey": "table.header.namespace",
    "resourceKey": "namespace",
    "textFunction": [Function],
    "tooltipKey": "table.header.application.namespace.tooltip",
    "transformFunction": [Function],
    "transforms": Array [
      [Function],
    ],
  },
  Object {
    "disabled": [Function],
    "msgKey": "table.header.clusters",
    "resourceKey": "clusterCount",
    "textFunction": [Function],
    "tooltipKey": "table.header.application.clusters.tooltip",
    "transformFunction": [Function],
  },
  Object {
    "disabled": [Function],
    "msgKey": "table.header.resource",
    "resourceKey": "hubChannels",
    "textFunction": [Function],
    "tooltipKey": "table.header.application.resource.tooltip",
    "transformFunction": [Function],
  },
  Object {
    "disabled": [Function],
    "msgKey": "table.header.timeWindow",
    "resourceKey": "hubSubscriptions",
    "textFunction": [Function],
    "tooltipKey": "table.header.application.timeWindow.tooltip",
    "transformFunction": [Function],
  },
  Object {
    "msgKey": "table.header.created",
    "resourceKey": "created",
    "transformFunction": [Function],
  },
]
`;

exports[`definitions/index tests should return the uri key of node 1`] = `"name"`;
